source/FreeRTOS_DNS_Parser.c

```diff
@@ -1144,7 +1146,7 @@
                         #endif
 
                         #if defined( ipconfigIPv4_BACKWARD_COMPATIBLE ) && ( ipconfigIPv4_BACKWARD_COMPATIBLE == 1 )
-                            if( xApplicationDNSQueryHook( xSet.pcName ) != pdFALSE )
+                            if( xApplicationDNSQueryHook( ( const char * ) ucNBNSName ) != pdFALSE )
```

Comment:
As you may have noticed, `Set.pcName` was used as a parameter in a different context. 
The function `DNS_TreatNBNS` as it is now was still not well tested. I have a new version in preparation, see PR #800.